File: BACK/Sistema-Escolar-crud_professor/Models/Turma.cs
using System;
using System.ComponentModel;
using Sistema_Escolar.Validations;

namespace API.Models
{
    public class Turma
    {
        public Turma () => CriadoEm = DateTime.Now;

        public int Id { get; set; }
        public int CodigoTurma{get; set;}
        public string ProfessorCpf{get;set;}
        public virtual Professor Professor { get; set; }
        public string disciplina {get; set;}
        public TimeSpan horario {get; set;}
        public double valor{get; set;}
        public DateTime CriadoEm { get; set; }
    }
}

File: FRONT/src/app/models/Turma.ts
import { Professor } from './../components/professor/professor.model';
export interface Turma {
    id?: number
    codigoTurma: number
    professorCpf: string
    disciplina: string
    horario: Date
    valor: number
    professorId: string
    prefessor?: Professor
}